Fix filewatcher on MacOS

After #13223, there are some failures in [FileWatcherTest](https://github.com/enso-org/enso/blob/develop/lib/scala/filewatcher/src/test/java/org/enso/filewatcher/FileWatcherTest.java) on MacOS x64.

Failures:
- https://github.com/enso-org/enso/actions/runs/15635806358/job/44063194801?pr=13212#step:14:2712
- https://github.com/enso-org/enso/actions/runs/15636914304/job/44091309827?pr=13256#step:14:2667
- https://github.com/enso-org/enso/actions/runs/15671145974/job/44144050173#step:14:2078
  - Engine Nightly Checks (JVM Tests on macos aarch64)

Note that `io.methvin:directory-watcher` uses JNA when listening on MacOS. See for example https://github.com/gmethvin/directory-watcher/blob/main/core/src/main/java/io/methvin/watchservice/MacOSXListeningWatchService.java

I am able to simulate the test failure on my Mac:
```
error] Caused by: java.lang.AssertionError: 
[error] Expected: is <WatcherEvent[path=/var/folders/36/74fmy6d51s9ct3v8lskdy07r0000gn/T/junit14909065946170544076/junit4413200291011904427/subdir/a.txt, eventType=CREATE]>
[error]      but: was <WatcherEvent[path=/var/folders/36/74fmy6d51s9ct3v8lskdy07r0000gn/T/junit14909065946170544076/junit4413200291011904427/subdir, eventType=MODIFY]>
[error]     at org.enso.filewatcher.FileWatcherTest.assertNextEventIs(FileWatcherTest.java:179)
[error]     at org.enso.filewatcher.FileWatcherTest.receiveModifyEventInSubdir(FileWatcherTest.java:132)
[error]     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[error]     at java.lang.reflect.Method.invoke(Method.java:565)
[error]     ...
[error] Caused by: java.lang.AssertionError: 
[error] Expected: is <WatcherEvent[path=/var/folders/36/74fmy6d51s9ct3v8lskdy07r0000gn/T/junit14909065946170544076/junit9877418823046181001/subdir/a.txt, eventType=CREATE]>
[error]      but: was <WatcherEvent[path=/var/folders/36/74fmy6d51s9ct3v8lskdy07r0000gn/T/junit14909065946170544076/junit9877418823046181001/subdir, eventType=MODIFY]>
[error]     ... 43 more
[error] Caused by: java.lang.AssertionError: 
[error] Expected: is <WatcherEvent[path=/var/folders/36/74fmy6d51s9ct3v8lskdy07r0000gn/T/junit14909065946170544076/junit7914778302638043969/subdir/a.txt, eventType=CREATE]>
[error]      but: was <WatcherEvent[path=/var/folders/36/74fmy6d51s9ct3v8lskdy07r0000gn/T/junit14909065946170544076/junit7914778302638043969/subdir, eventType=MODIFY]>
[error]     ... 43 more
[info] Test run org.enso.filewatcher.FileWatcherTest finished: 4 failed, 0 ignored, 8 total, 76.332s
[error] Failed: Total 8, Failed 4, Errors 0, Passed 4
```

--- a/lib/scala/filewatcher/src/test/java/org/enso/filewatcher/FileWatcherTest.java
+++ b/lib/scala/filewatcher/src/test/java/org/enso/filewatcher/FileWatcherTest.java
@@ -22,8 +22,13 @@ import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
 
 public class FileWatcherTest {
+  private static final Logger LOGGER = LoggerFactory.getLogger(FileWatcherTest.class);
+  private static final Level AT = Level.DEBUG;
   private static final long TIMEOUT_SECONDS = 5;
 
   @Rule public RetryTestRule retryRule = new RetryTestRule(3);
@@ -175,8 +180,15 @@ public class FileWatcherTest {
   }
 
   private void assertNextEventIs(WatcherEvent expectedEvent) {
-    var event = pollEvent();
-    assertThat(event, is(expectedEvent));
+    LOGGER.atLevel(AT).log("Expecting event: {}", expectedEvent);
+    for (; ; ) {
+      var event = pollEvent();
+      LOGGER.atLevel(AT).log("  got event: {}" + event);
+      if (expectedEvent.equals(event)) {
+        LOGGER.atLevel(AT).log("  good!");
+        return;
+      }
+    }
   }
 
   private WatcherEvent pollEvent() {

